Cannot delete message/thread from the thread menu

### Expected behavior
 message/thread is deleted from the thread menu


### Actual behavior
After i click delete on the thread menu, the message is gone in the envelope list but the opened message on the thread doesnt go away. No error message. Then if i refresh right away, This error is shown:` [ERROR] mail: could not load envelope thread {app: "mail", uid: "greta", threadId: 13448, error: Error: Request failed with status code 403`

And the thread keeps loading and nothing happens.


### Mail app

**Mail app version:** 1.9

--- a/src/store/mutations.js
+++ b/src/store/mutations.js
@@ -221,6 +221,16 @@ export default {
 					list.splice(idx, 1)
 				}
 			})
+
+		// Delete references from other threads
+		for (const [key, env] of Object.entries(state.envelopes)) {
+			if (!env.thread) {
+				continue
+			}
+
+			const thread = env.thread.filter(threadId => threadId !== id)
+			Vue.set(state.envelopes[key], 'thread', thread)
+		}
 	},
 	addMessage(state, { message }) {
 		Vue.set(state.messages, message.databaseId, message)

--- a/src/tests/unit/store/mutations.spec.js
+++ b/src/tests/unit/store/mutations.spec.js
@@ -824,6 +824,12 @@ describe('Vuex store mutations', () => {
 					id: 123,
 					uid: 12345,
 				},
+				12346: {
+					mailboxId: 27,
+					id: 123,
+					uid: 12345,
+					thread: [12345, 12346],
+				},
 			},
 			mailboxes: {
 				27: {
@@ -869,6 +875,12 @@ describe('Vuex store mutations', () => {
 					mailboxId: 27,
 					id: 123,
 					uid: 12345,
+				},
+				12346: {
+					mailboxId: 27,
+					id: 123,
+					uid: 12345,
+					thread: [12346],
 				},
 			},
 			mailboxes: {

